src/components/SearchPanel.js

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import withStyles from '@material-ui/core/styles/withStyles';
+import Paper from '@material-ui/core/Paper';
+import Grid from '@material-ui/core/Grid';
+import ListItem from '@material-ui/core/ListItem';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import Close from '@material-ui/icons/Close';
+import SearchInputField from './SearchInputField';
+
+const styles = {
+    listItemText: {
+        '&:first-child': {
+            paddingLeft: 30,
+            paddingTop: 0,
+            paddingBottom: 0
+        }
+    },
+    paper: {
+        backgroundColor: '#f5f5f5',
+        position: 'relative',
+        zIndex: -1
+    },
+    menuItems: {
+        fontSize: '12px',
+        lineHeight: 2
+    },
+    closeButton: {
+        marginRight: '10px',
+        marginTop: '10px',
+        position: 'absolute',
+        top: 0,
+        right: 0,
+        zIndex: 1
+    },
+    searchInputField: {
+        float: 'right'
+    }
+};
+
+function SearchPanel({ menu, classes, close }) {
+    const [searchTerm, setSearchTerm] = useState();
+
+    // flatten out the menu object into a single array of key value pairs
+    const menuEntries = menu
+        .map(s => s.columns)
+        .flat()
+        .map(c => c.categories)
+        .flat()
+        .map(i => i.items)
+        .flat();
+
+    const uniqueEntries = Object.values(
+        menuEntries.reduce((c, e) => {
+            if (!c[e.title]) {
+                return { ...c, [e.title]: e };
+            }
+            return c;
+        }, {})
+    );
+
+    const handleFieldChange = (propertyName, newValue) => {
+        setSearchTerm(newValue);
+    };
+    return (
+        <Paper classes={{ root: classes.paper }}>
+            <Button onClick={close} color="secondary" style={styles.closeButton}>
+                <Close />
+            </Button>
+            <Grid container>
+                <Grid item xs={12} sm={6} md={4} lg={3} xl={3} justify-content="flex-end">
+                    <SearchInputField
+                        value={searchTerm}
+                        onChange={handleFieldChange}
+                        textFieldProps={{
+                            autoFocus: true
+                        }}
+                        placeholder="start typing..."
+                    />
+                    {searchTerm &&
+                        uniqueEntries
+                            .filter(e => e.title.toLowerCase().startsWith(searchTerm.toLowerCase()))
```

Comment:
Could there be a way  to search within the words as well? So searcing for 'accounts' would bring back sales accounts etc

Reply:
Probably... filtering a big list on a .`includes()` is a lot more expensive than `.startsWith()` so I went with the latter. I could look into a debounce but I really like instantaneous feel of the search in its current form... Maybe `useMemo()` can help.